p2panda-rs/src/instance/instance.rs

```diff
@@ -1,23 +1,25 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 
 //! Types and methods for deriving and maintaining materialised documents.
-use std::{collections::HashMap, convert::TryFrom};
+use std::collections::btree_map::Iter;
+use std::collections::BTreeMap;
+use std::convert::TryFrom;
 
 use crate::instance::error::InstanceError;
-use crate::operation::{AsOperation, Operation, OperationValue};
+use crate::operation::{AsOperation, Operation, OperationValue, OperationWithMeta};
 
 /// The materialised view of a reduced collection of `Operations` describing a document.
 #[derive(Debug, PartialEq, Default)]
-pub struct Instance(HashMap<String, OperationValue>);
+pub struct Instance(BTreeMap<String, OperationValue>);
 
 impl Instance {
     /// Returns a new `Instance`.
     fn new() -> Self {
-        Self(HashMap::new())
+        Self(BTreeMap::new())
```

Comment:
I think instances should have a `document: Hash` so we can keep track of which document they are describing no?

Reply:
The flow has changed slightly now. `DocumentBuilder` now returns `Documents` which have a `view` field holding an already resolved view of this document. They also have `id` and `schema` fields. :+1: in any case, we have access to document id hash now.